tap_mysql/discover_utils.py

```diff
@@ -37,16 +45,98 @@
     'bigint': 8
 }
 
-FLOAT_TYPES = {'float', 'double'}
+BOOL_TYPES = {'bit'}
+
+JSON_TYPES = {'json'}
+
+FLOAT_TYPES = {'float', 'double', 'decimal'}
 
-DATETIME_TYPES = {'datetime', 'timestamp', 'date'}
+DATETIME_TYPES = {'datetime', 'timestamp', 'time', 'date'}
 
 BINARY_TYPES = {'binary', 'varbinary'}
 
 SPATIAL_TYPES = {'geometry', 'point', 'linestring',
                  'polygon', 'multipoint', 'multilinestring',
                  'multipolygon', 'geometrycollection'}
 
+# A set of all supported column types listed above
+SUPPORTED_COLUMN_TYPES_AGGREGATED = \
+    STRING_TYPES \
+        .union(FLOAT_TYPES) \
+        .union(DATETIME_TYPES) \
+        .union(BINARY_TYPES) \
+        .union(SPATIAL_TYPES) \
+        .union(BOOL_TYPES) \
+        .union(JSON_TYPES) \
+        .union(BYTES_FOR_INTEGER_TYPE.keys())
+
+
+def is_supported_column_type(column_sql_type: str) -> bool:
+    """
+    Checks if the given sql type is supported
+
+    Args:
+        column_sql_type: Column sql data type from the catalog metadata
+            could be in the format {TYPE}(size) or {TYPE}, e.g bigint(100), timestamp ..etc
```

Comment:
is this compatible with column modifiers like `unsigned` ?

Reply:
Hmm, if the sql type has length then usually `unsigned` comes after it, so it will be filtered out and only the data type remains. Otherwise, we gonna be in trouble 😅. So probably using data type as you suggested above would save us.

Reply:
This is not outdated